ledger-service/http-json/src/main/scala/com/digitalasset/http/endpoints/MeteringReportEndpoint.scala

```diff
@@ -0,0 +1,262 @@
+    fields.get(name) match {
+      case Some(JsString(txt)) => Timestamp.fromString(txt).map(Some.apply)
+      case _ => Right(None)
+    }
+  }
+
+  def toOptionalApplicationId(fields: Map[String, JsValue] , name: String): Either[String, Option[ApplicationId]] = {
+    fields.get(name) match {
+      case Some(JsString(txt)) => ApplicationId.fromString(txt).map(Some.apply)
+      case _ => Right(None)
+    }
+  }
+
+  def toApplicationId(fields: Map[String, JsValue] , name: String): Either[String, ApplicationId] = {
+    fields.get(name) match {
+      case Some(JsString(txt)) => ApplicationId.fromString(txt)
+      case _ => Left(s"Expected ApplicationId $name")
+    }
+  }
+
+  def toParticipantId(fields: Map[String, JsValue] , name: String): Either[String, ParticipantId] = {
+    fields.get(name) match {
+      case Some(JsString(txt)) => ParticipantId.fromString(txt)
+      case _ => Left(s"Expected ParticipantId $name")
+    }
+  }
+
+  def toLong(fields: Map[String, JsValue], name: String): Either[String, Long] = {
+    fields.get(name) match {
+      case Some(JsNumber(value)) => Right(value.toLong)
+      case _ => Left(s"Expected JsNumber $name")
+    }
+  }
+
+  def toBoolean(fields: Map[String, JsValue], name: String): Either[String, Boolean] = {
+    fields.get(name) match {
+      case Some(JsBoolean(value)) => Right(value)
+      case _ => Left(s"Expected JsBoolean $name")
+    }
+  }
+
+  def toArray(fields: Map[String, JsValue], name: String): Either[String, Vector[JsValue]] = {
+    fields.get(name) match {
+      case Some(JsArray(values)) => Right(values)
+      case _ => Left(s"Expected JsBoolean $name")
+    }
+  }
+
+  def toObject(fields: Map[String, JsValue], name: String): Either[String, JsObject] = {
+    fields.get(name) match {
+      case Some(j@JsObject(_)) => Right(j)
+      case _ => Left(s"Expected JsObject $name")
+    }
+  }
+
+
+  implicit object MeteringReportRequestFormat extends RootJsonFormat[MeteringReportRequest] {
+
+    def write(request: MeteringReportRequest): JsObject = {
+      import request._
+      JsObject(
+        "from" -> JsString(from.toString),
+        "to" -> to.fold[JsValue](JsNull){t => JsString(t.toString)},
+        "application" -> applicationId.fold[JsValue](JsNull){a => JsString(a)},
+      )
+    }
+
+    def read(value: JsValue): MeteringReportRequest = {
+      val fields: Map[String, JsValue] = value.asJsObject.fields
+      (for {
+        from <- toTimestamp(fields, "from")
+        to <- toOptionalTimestamp(fields, "to")
+        applicationId <- toOptionalApplicationId(fields, "application")
+      } yield {
+        MeteringReportRequest(from, to, applicationId)
+      }) match {
+        case Right(request) => request
+        case Left(message) => throw DeserializationException(s"MeteringReportRequest parse problem ($message) parsing: $value")
+      }
+    }
+  }
```

Comment:
```suggestion
  implicit val MeteringReportRequestFormat: RootJsonFormat[MeteringReportRequest] =
    jsonFormat2(MeteringReportRequest.apply)
```

Reply:
The target json has to match the published format, see:
https://docs.google.com/document/d/1HQyVLCANqw_l_gbxBzgOqsiwmIDNlwCRQffVYJnYTeU/edit#heading=h.1mjra845v4le

This change, in isolation, would not do that.

Using the explicit JsonFormat makes it much harder to accidentally change the Json output as a side effect of renaming a field.

Reply:
In combination with the field renames this can work if that makes for easier maintenance.

Reply:
Done.